src/System.Private.TypeLoader/src/Internal/TypeSystem/RuntimeNoMetadataType.cs

```diff
@@ -280,6 +283,69 @@ public override TypeDesc UnderlyingType
             }
         }
 
+        private void GetTypeNameHelper(out string name, out string nsName, out string assemblyName)
+        {
+            TypeReferenceHandle typeRefHandle;
+            QTypeDefinition qTypeDefinition;
+            MetadataReader reader;
+
+            RuntimeTypeHandle genericDefinitionHandle = GetTypeDefinition().GetRuntimeTypeHandle();
+            Debug.Assert(!genericDefinitionHandle.IsNull());
+
+            // NOTE: We're ignoring the containing type for compatiblity with SigFormat.cpp
```

Comment:
@fadimounir What does this comment mean? This helper gets the name and the namespace of a type from metadata. It doesn't do any formatting. Containing type name is naturally not part of any of this. Why would we need to be sigformat compatible in any way?